dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/apphub/AppHubUtils.java

```diff
@@ -86,6 +89,32 @@ public static void validateApiVersion( String apiVersion )
         }
     }
 
+    /**
+     * Validates the UUID. Apphub uses UUID-v4 for all identifiers.
+     *
+     * @param uuid the uuid
+     * @throws IllegalQueryException if the UUID is invalid
+     */
+
+    public static void validateUuid( String uuid )
```

Comment:
Usually we return the error codes from the service layer, for example, you can rewrite this to: "public static boolean validUuid( String uuid )"; Then return false instead of an exception

Reply:
If there are different possible error states there is merit in either returning the specific `ErrorCode` or throw an exception, so that the caller can understand exactly what the issue is.

Reply:
The IllegalQueryException will be caught in the web layer, and we have a similar approach for analytics (see `DefaultQueryValidator`).